api/oc_message_internal.h

```diff
@@ -57,15 +57,34 @@ oc_message_t *oc_message_allocate_outgoing(void);
  */
 oc_message_t *oc_message_allocate_outgoing_with_size(size_t size);
 
+/**
+ * @brief Get maximum size of the message buffer
+ *
+ * The size of the buffer depends on compilation options (it may be static or
+ * dynamic and have differing sizes).
+ *
+ * @return size_t size of the message buffer
+ */
+size_t oc_message_max_buffer_size(void);
+
 /**
  * @brief Get size of the message buffer
  *
  * The size of the buffer depends on compilation options (it may be static or
  * dynamic and have differing sizes).
  *
+ * @param message the message to get the buffer size
  * @return size_t size of the message buffer
  */
-size_t oc_message_buffer_size(void);
+size_t oc_message_buffer_size(const oc_message_t *message);
```

Comment:
I would add `OC_NONNULL()` to `oc_message_buffer_size` and `oc_message_shrink_buffer` and remove the 

```C
  if (message == NULL) {
    ...
  }
```

checks. Tthese are internal functions which are never used with NULL, we don't have to worry about user inputs, so compiler checks are sufficient.

Reply:
Fixed